go/tools/go-upgrade/go-upgrade.go

```diff
@@ -384,7 +385,7 @@ func replaceGoVersionInCodebase(old, new *version.Version, workflowUpdate bool)
 
 	if !isSameMajorMinorVersion(old, new) {
 		err = replaceInFile(
-			[]*regexp.Regexp{regexp.MustCompile(`go[[:space:]]*([0-9.]+)`)},
```

Comment:
Shouldn't it be:
```
]*regexp.Regexp{regexp.MustCompile(`go[[:space:]]*([0-9.]+).([0-9.]+)`)},
```

Aren't we parsing this e.g.: `go1.20.6`. The * means any number (0 or more) spaces. Without that, it's expected to have one space.

Reply:
We are parsing the `go 1.20` at the beginning of the `go.mod`` file